File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.cs
public class Solution {
    public int MinDeletions(string s) {
            var f = new int[26];
            for (int i = 0; i < s.Length; i++)
            {
                f[s[i] - 'a']++;
            }
            var h = new HashSet<int>();
            int res = 0;
            for (int i = 0; i < 26; i++)
            {
                if (!h.Contains(f[i]))
                {
                    h.Add(f[i]);
                }
                else
                {
                    while (f[i]>0 && h.Contains(f[i]))
                    {
                        res++;
                        f[i]--;
                    }
                    h.Add(f[i]);
                }
            }
            return res;
    }
}

File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.cpp
class Solution {
public:
    int minDeletions(string s) {
        map<char,int>m;
        for(auto i:s){
            m[i]++;
        }
        priority_queue<int> q;
        for(auto i:m){
            q.push(i.second);
        }
        int res=0;
        while(q.size()>1){
            int a = q.top();
            q.pop();
            if(a==q.top()){
                if(a-1>0){
                    q.push(a-1);
                }
                res++;
            }
        }
        return res;
    }
};